### PR TITLE
Display stickers

### PR DESCRIPTION
**Pull request by Tyuoli**
https://github.com/vector-im/riot-meta/issues/146
https://github.com/vector-im/riot-android/issues/2097

### CODE REVIEW

**Comment by ylecollen** [comment]
1- StickerInfo could have inherited from VideoInfo or ImageInfo (i did not check which one match)
2- StickerMessage  could have inherited from VideoMessage or ImageMessage
With these both inheritages, you did not need to c+p to manage the upload + download on application.
it would have saved a lot of pains.

3- in 
 public StickerMessage  () {
        super() <--- you did not set the type, see imageMessage
    }

**Comment by Tyuoli** [comment]
m.sticker isn't a new message type but a new event type. I will try to create a new  StickerMessageModel to parse sticker json and inherit of ImageMessage.

File: matrix-sdk/src/main/java/org/matrix/androidsdk/MXSession.java [hunk 1]
```diff
@@ -741,6 +743,16 @@ public void removeMediasBefore(final Context context, final long timestamp) {
                                     filesToKeep.add(Uri.parse(mediaMessage.getUrl()).getPath());
                                 }
                             }
+                        } else if (TextUtils.equals(Event.EVENT_TYPE_STICKER, event.getType())) {
+                            StickerMessage stickerMessage = JsonUtils.toStickerMessage(event.getContent());
+
+                            if (stickerMessage.isThumbnailLocalContent()) {
```
Comment (manuroe) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/MXSession.java]: As `StickerMessage` is now a `MediaMessage`,  could we merge this C+P code with the code above?

File: matrix-sdk/src/main/java/org/matrix/androidsdk/rest/model/message/Message.java [hunk 2]
```diff
@@ -26,6 +26,10 @@
     public static final String MSGTYPE_FILE = "m.file";
     public static final String FORMAT_MATRIX_HTML = "org.matrix.custom.html";
 
+    // Add, in local, a fake message type in order to StickerMessage can inherit Message class
+    // Because sticker isn't a new message type but a new event type without msgtype field
```
Comment (manuroe) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/rest/model/message/Message.java]: Can we remove "new" in the comment?
Stickers will be no more new next time we play(code) with them.

File: matrix-sdk/src/main/java/org/matrix/androidsdk/util/EventDisplay.java [hunk 3]
```diff
@@ -106,6 +107,8 @@ public CharSequence getTextualDisplay() {
     public CharSequence getTextualDisplay(Integer displayNameColor) {
         CharSequence text = null;
 
+        Log.e(LOG_TAG, "#getTextualDisplay : " + mEvent.eventId);
```
Comment (manuroe) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/util/EventDisplay.java]: Remove this dev log.

File: matrix-sdk/src/main/java/org/matrix/androidsdk/rest/model/message/StickerJsonMessage.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.matrix.androidsdk.rest.model.message;
+
+public class StickerJsonMessage {
```
Comment (manuroe) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/rest/model/message/StickerJsonMessage.java]: Could we have a comment for this class?
Explaining that it is just here as an intermediate object to create a StickerMessage from a m.sticker event type.

File: matrix-sdk/src/main/java/org/matrix/androidsdk/MXSession.java [hunk 5]
```diff
@@ -727,7 +729,7 @@ public void removeMediasBefore(final Context context, final long timestamp) {
             if (null != events) {
                 for (Event event : events) {
                     try {
-                        if (TextUtils.equals(Event.EVENT_TYPE_MESSAGE, event.getType())) {
+                        if (TextUtils.equals(Event.EVENT_TYPE_MESSAGE, event.getType()) || TextUtils.equals(Event.EVENT_TYPE_STICKER, event.getType())) {
```
Comment (manuroe) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/MXSession.java]: Does it work?

**Pull request merged**